apps/api/v1/test/lib/selected-calendars/_post.test.ts

```diff
@@ -0,0 +1,122 @@
+import prismaMock from "../../../../../../tests/libs/__mocks__/prismaMock";
+
+import type { Request, Response } from "express";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { createMocks } from "node-mocks-http";
+import { describe, expect, test } from "vitest";
+
+import { HttpError } from "@calcom/lib/http-error";
+
+import handler from "../../../pages/api/selected-calendars/_post";
+
+type CustomNextApiRequest = NextApiRequest & Request;
+type CustomNextApiResponse = NextApiResponse & Response;
+
+describe("POST /api/selected-calendars", () => {
+  describe("Errors", () => {
+    test("Returns 403 if non-admin user tries to set userId in body", async () => {
+      const { req, res } = createMocks<CustomNextApiRequest, CustomNextApiResponse>({
+        method: "POST",
+        body: {
+          integration: "google",
+          externalId: "ext123",
+          userId: 444444,
+        },
+      });
+
+      req.userId = 333333;
+      req.isSystemWideAdmin = false;
```

Comment:
For a non admin user you don't need to explicitly set `req.isSystemWideAdmin = false`, you can remove it.